community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -600,7 +598,8 @@ private void commit() throws TransactionFailureException
                     }
                 }
 
-                context.init( locks );
+                statementLocks.prepareForCommit();
```

Comment:
Sequence of operation is not really clear here, java docs does not anything about optimistic locks that will be taken here or why we init context with pessimistic lock only